packages/components/src/components/file/file.component.ts

```diff
@@ -300,6 +301,56 @@ export default class SynFile extends SynergyElement implements SynergyFormContro
     this.files = files;
   }
 
+  private async handleTransferItems(items: DataTransferItemList | null): Promise<FileList> {
+    if (!items || items.length !== 1) {
+      this.value = '';
+      return new Promise((_resolve, reject) => { reject(new Error('No proper items found')); });
+    }
+
+    const itemsArray = Array.from(items);
+
+    const entries: (FileSystemEntry | null)[] = itemsArray.map((item) => item.webkitGetAsEntry());
+
+    if (!entries[0]?.isDirectory) {
+      this.value = '';
+      return new Promise((_resolve, reject) => { reject(new Error('Not a folder')); });
```

Comment:
What are we doing with the rejection here? We do not throw `syn-invalid` or another event to the outside as far as I can see. Also, we don´t check for rejection in the calling method.